src/Console/InfectionContainer.php

```diff
@@ -213,24 +212,33 @@ public static function create(): self
         ]);
     }
 
-    public function buildDynamicDependencies(InputInterface $input): void
-    {
-        $this['infection.config'] = static function (self $container) use ($input): InfectionConfig {
+    public function withDynamicParameters(
+        ?string $configFile,
```

Comment:
This is a lot of parameters... which may be why it was so conveniently hidden behind `Input`. However I think that:

- It is a hideous hack: `Input` is a horrible API to work with in general (as most UI adapters), you need to figure out the origin (argument or option), if the argument/option has a value, the type of the value, normalizing the value... It doesn't make sense to leak this API here in the container IMO
- It hides the complexity which in this case should show a deeper problem: we shouldn't have that many dynamic services. That said for now it is my opinion, I plan to look into it whether or not we can reduce that number, but it may be that I'm just overly optimistic about it as well